src/Symfony/Component/DependencyInjection/Tests/Compiler/ServiceLocatorTagPassTest.php

```diff
@@ -143,4 +146,56 @@ public function testBindingsAreCopied()
         $this->assertSame(['foo'], array_keys($locator->getBindings()));
         $this->assertInstanceOf(BoundArgument::class, $locator->getBindings()['foo']);
     }
+
+    public function testIndexedByServiceIdWithDecoration()
+    {
+        $container = new ContainerBuilder();
+
+        $locator = new Definition(Locator::class);
+        $locator->setPublic(true);
+        $locator->addArgument(new ServiceLocatorArgument(new TaggedIteratorArgument('test_tag', null, null, true)));
+
+        $container->setDefinition(Locator::class, $locator);
+
+        $service = new Definition(Service::class);
+        $service->setPublic(true);
+        $service->addTag('test_tag');
+
+        $container->setDefinition(Service::class, $service);
+
+        $decorated = new Definition(Decorated::class);
```

Comment:
`Decorated::class` is not loaded by this test. That is, running this test in isolation will fail because the class cannot be found. It relies on running the entire test suite in alphabetical order so that `AutowireRequiredMethodsPassTest` (which runs before it) will load it on its behalf.

Reply:
Good catch, see #48654